### PR TITLE
Multiple voting systems: Two-axis voting

### PR DESCRIPTION
**Pull request by jimrandomh**
Adds two-axis voting. Should all be working, although the frontend UI isn't polished.

### CODE REVIEW

**Comment by michaelkeenan** [comment]
Thanks for doing this!

I found that the first agree-axis vote didn't update the score in the client, and the score only updated when the server request returned. That's because setVoteClient only calls addVoteClient if voteType isn't null. That could perhaps be changed to something like:

```if (voteType !== null || _.values(extendedVote).some((val) => val !== 'neutral')) {
    document = addVoteClient({document, collection, voteType, extendedVote, user, votingSystem});
}```

Though, that's not quite right because a non-voted Overall voteType can be null or 'neutral', depending on what was clicked in what order. I *think* the solution is to set voteType to 'neutral' rather than null in OverallVoteButton.tsx and AddVoteButton.tsx, and/or set a default value for the voteType in withVote's vote function, and then change that conditional to check for non-'neutral' voteTypes rather than non-null ones, but idk, I'm new to this stack and codebase. Hope that made sense!

**Comment by jpaddison3** [comment]
Dumb question from JP: What's the difference between this and #4355 ?

**Comment by darkruby501** [comment]
>Dumb question from JP: What's the difference between this and #4355 ?

@jpaddison3 Not a dumb question. They're trying to do the same thing, except one is the version Keenan and I made and the other is Jim's version. After it had taken us a few weeks, I said to Jim "I remember when I thought I could do this in three days." Jim responded that he could do it in three days. Keenan and I encouraged him to try.

File: packages/lesswrong/lib/voting/votingSystems.tsx [hunk 1]
```diff
@@ -0,0 +1,123 @@
...
+      extendedScore: null,
+    };
+  }
+});
+
+registerVotingSystem({
+  name: "twoAxis",
+  description: "Two-Axis Approve and Agree",
+  getCommentVotingComponent: () => Components.TwoAxisVoteOnComment,
+  addVoteClient: (document: VoteableTypeClient, extendedVote: any, currentUser: UsersCurrent): any => {
```
[the remark this answers is not in the document]
Comment (michaelkeenan) [reply packages/lesswrong/lib/voting/votingSystems.tsx:53]: Wait, it doesn't rerun the new power, but the new score.

File: packages/lesswrong/components/votes/VoteButton.tsx [hunk 2]
```diff
@@ -63,32 +60,35 @@ const styles = (theme: ThemeType): JssStyles => ({
...
 }) => {
-  const currentUser = useCurrentUser();
-  const { openDialog } = useDialog();
-  const { captureEvent } = useTracking();
   const [votingTransition, setVotingTransition] = useState<any>(null);
   const [bigVotingTransition, setBigVotingTransition] = useState(false);
   const [bigVoteCompleted, setBigVoteCompleted] = useState(false);
+  
+  const wrappedVote = (strength: "big"|"small"|"neutral") => {
+    if (currentStrength === "small")
```
Comment (michaelkeenan) [inline packages/lesswrong/components/votes/VoteButton.tsx:87]: This line prevents bigVotes, changing them to neutral votes. I think it should be `if (strength === currentStrength)`

**Comment by darkruby501** [comment]
Just in case you missed it:
Screen Shot 2021-12-15 at 6 16 41 AM

**Comment by darkruby501** [comment]
Clicking "move to alignment forum" or "remove from alignment forum" causes all comments to disappear. This is a bug in production, but coming up in testing.

File: packages/lesswrong/components/votes/OverallVoteAxis.tsx [hunk 3]
```diff
@@ -0,0 +1,162 @@
...
+    fontStyle: "italic"
+  }
+})
+
+const OverallVoteAxis = ({ document, hideKarma=false, voteProps, classes, showBox=false }: {
+  document: VoteableTypeClient,
+  hideKarma?: boolean,
+  voteProps: VotingProps<VoteableTypeClient>,
+  classes: ClassesType,
+  showBox: boolean
```
Comment (jimrandomh) [inline packages/lesswrong/components/votes/OverallVoteAxis.tsx:58]:
```suggestion
  showBox?: boolean
```
This has a default value in the argument destructuring, but isn't marked as having an optional type. Changing the `:` to `?:` makes it properly optional.

Comment (michaelkeenan) [reply packages/lesswrong/components/votes/OverallVoteAxis.tsx:58]: Good to know, thanks!

**Review by darkruby501 — APPROVED** [review]

**Pull request merged**